Python/pythonrun.c

```diff
@@ -1535,12 +1063,42 @@ _PyErr_Display(PyObject *file, PyObject *unused, PyObject *value, PyObject *tb)
         }
     }
 
+    if (!value) {
+        goto fallback;
+    }
+
+    // Try first with the traceback.
+    int unhandled_keyboard_interrupt = _PyRuntime.signals.unhandled_keyboard_interrupt;
+    PyObject *traceback_module = PyImport_ImportModule("traceback");
+
+    if (traceback_module == NULL) {
+        goto fallback;
+    }
+
+    PyObject *print_tb_func = PyObject_GetAttrString(traceback_module, "_print_exception_bltin");
+
+    if (print_tb_func == NULL || !PyCallable_Check(print_tb_func)) {
+        Py_DECREF(traceback_module);
+        goto fallback;
+    }
+
+    PyObject* result = PyObject_CallOneArg(print_tb_func, value);
+
+    Py_DECREF(traceback_module);
+    Py_XDECREF(print_tb_func);
+    _PyRuntime.signals.unhandled_keyboard_interrupt = unhandled_keyboard_interrupt;
```

Comment:
`_PyRuntime.signals.unhandled_keyboard_interrupt = unhandled_keyboard_interrupt;`  is not done if we goto fallback above.